internal/api/provider/oidc.go

```diff
@@ -200,6 +204,127 @@ func parseLinkedinIDToken(token *oidc.IDToken) (*oidc.IDToken, *UserProvidedData
 	return token, &data, nil
 }
 
+type AzureIDTokenClaims struct {
+	jwt.StandardClaims
+
+	Email                              string `json:"email"`
+	PreferredUsername                  string `json:"preferred_username"`
+	XMicrosoftEmailDomainOwnerVerified any    `json:"xms_edov"`
+}
+
+func (c *AzureIDTokenClaims) IsEmailVerified() bool {
+	// If xms_edov is not set, and an email is present or xms_edov is true,
+	// only then is the email regarded as verified.
+	// https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-off-email-claim-authorization#using-the-xms_edov-optional-claim-to-determine-email-verification-status-and-migrate-users
+	emailVerified := false
+
+	edov := c.XMicrosoftEmailDomainOwnerVerified
+
+	if edov == nil {
+		// An email is provided, but xms_edov is not -- probably not
+		// configured, so we must assume the email is verified as Azure
+		// will only send out a potentially unverified email address in
+		// single-tenanat apps.
+		emailVerified = c.Email != ""
+	} else {
+		edovBool := false
+
+		// Azure can't be trusted with how they encode the xms_edov
+		// claim. Sometimes it's "xms_edov": "1", sometimes "xms_edov": true.
+		switch edov.(type) {
+		case bool:
+			edovBool = edov.(bool)
+
+		case string:
+			edovBool = edov.(string) == "1" || edov.(string) == "true"
+
+		case int:
+			edovBool = edov.(int) != 0
+
+		case int32:
+			edovBool = edov.(int32) != 0
+
+		case int64:
+			edovBool = edov.(int64) != 0
+
+		case float32:
+			edovBool = edov.(float32) != 0
+
+		case float64:
+			edovBool = edov.(float64) != 0
+		}
+
+		emailVerified = c.Email != "" && edovBool
```

Comment:
Maybe now, but I don't trust Microsoft at all on this. I'd rather we go through all possible cases and never think about it again.

Reply:
hmm, i'd actually prefer if we know that they changed the API rather than let it "fail" softly 

maybe we can just 
```suggestion
		switch edov.(type) {
		case bool:
			edovBool = edov.(bool)

		case string:
			edovBool = edov.(string) == "1" || edov.(string) == "true"

		default:
		        edovBool = false
		}

		emailVerified = c.Email != "" && edovBool
```